sql/driver/doc.go

```diff
@@ -0,0 +1,284 @@
+	- ssl: A boolean. Enables/disables SSL connections. Defaults: false. Example: ssl=true
+	- ssl.ca.path: The path to the PEM file for the certificate authority. Implies ssl=true. Example: ssl.ca.path=/etc/ssl/ca.pem
+	- ssl.cert.path: The path to the TLS certificate. Implies ssl=true. Example: ssl.cert.path=/etc/ssl/cert.pem
+	- ssl.key.path: The path to the certificate key. Implies ssl=true. Example: ssl.key.path=/etc/ssl/key.pem
+	- ssl.key.password: The optional certificate password. Example: ssl.key.password=m0us3
+
+Some items in the client configuration cannot be set in the DSN, such as serialization factories and SSL configuration.
+You can use the following functions to set those configuration items globally:
+
+	- SetSerializationConfig(...)
+	- SetLoggerConfig(...)
+	- SetSSLConfig(...)
+
+Note that, these functions affect only the subsequent sql.Open calls, not the previous ones.
+
+Here's an example:
+
+	sc1 := &serialization.Config{}
+	sc2 := &serialization.Config{}
+	// no serialization configuration is used for the call below
+	db1, err := sql.Open("hazelcast", "")
+	// the following two sql.Open calls use sc1
+	err = driver.SetSerializationConfig(sc1)
+	db2, err := sql.Open("hazelcast", "")
+	db3, err := sql.Open("hazelcast", "")
+	// the following sql.Open call uses sc2
+	err = driver.SetSerializationConfig(sc2)
+	db4, err := sql.Open("hazelcast", "")
+
+Creating a Driver Instance Using driver.Open
+
+It is possible to create a driver instance using an existing Hazelcast client configuration using the driver.Open function.
+All client configuration items, except listeners are supported.
+
+	cfg := hazelcast.Config{}
+	cfg.Cluster.Name = "prod"
+	cfg.Serialization.SetPortableFactories(&MyPortableFactory{})
+	db := driver.Open(cfg)
+
+Executing Queries
+
+database/sql package supports two kinds of queries: The ones returning rows (select statements and a few others) and the rest (insert, update, etc.).
+The former kinds of queries are executed with QueryXXX methods and the latter ones are executed with ExecXXX methods of the sql.DB instance returned from sql.Open or driver.Open
+
+Use the question mark (?) for placeholders.
+
+Here is an Exec example:
+
+	q := `INSERT INTO person(__key, age, name) VALUES (?, ?, ?)`
+	result, err := db.Exec(q, 1001, 35, "Jane Doe")
+	// handle the error
+	cnt, err := result.AffectedRows()
+	// handle the error
+	fmt.Printf("Affected rows: %d\n", cnt)
+
+Note that LastInsertId is not supported and at the moment AffectedRows always returns 0.
+
+An example Query call:
+
+	q :=`SELECT name, age FROM person WHERE age >= ?`
+	rows, err := db.Query(q, 30)
+	// handle the error
+	defer rows.Close()
+	var name string
+	var age int
+	for rows.Next() {
+		err := rows.Scan(&name, &age)
+		// handle the error
+		fmt.Println(name, age)
+	}
+
+Context variants of Query and Exec, such as QueryContext and ExecContext are fully supported.
+Context variants are used to pass Hazelcast specific parameters.
```

Comment:
What does this mean? I did not get it. 

EDIT: I understood this after reading Passing Hazelcast Specific Parameters section. Maybe we can add a reference(or link) to that section here.

Reply:
Cross-references are not supported in Go, but I will try to explain in that better.

Reply:
Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/709/commits/b0798c9d652c2a57683c730b872d5d2fc19d71c0

Reply:
By adding a "reference" I did not mean a link. We can say that it is explained in the `Passing Hazelcast Specific Parameters` section below. 

Like this: 

They can be used to pass Hazelcast specific parameters, such as the cursor buffer size. **For details of how this works, check out Passing Hazelcast Specific Parameters section below.**

Reply:
OK, updated at: https://github.com/hazelcast/hazelcast-go-client/pull/709/commits/fc11615214789dfe475c36ea8deff6d4400e1ccc